### DISCUSSION TITLE
Multiple stages, same api gw id

### DISCUSSION THREAD

**Discussion by omar-dulaimi** [serverless/serverless#9174] [Q&A]
Hello,

I'm deploying multiple stages and have noticed that each deployment is created with a different api gateway id resulting in a different url for each stage.

We would like to have all of our stages on the same url:

```
https://123.execute-api.us-east-1.amazonaws.com/production/graphql
https://123.execute-api.us-east-1.amazonaws.com/staging/graphql
```

Is that possible?

I couldn't find any docs about that.

**Comment by pgrzesik** [comment]
Hello @omar-dulaimi - I believe the best approach would be to use a shared gateway in a similar approach as here: https://www.serverless.com/framework/docs/providers/aws/events/apigateway/#share-api-gateway-and-api-resources but instead of separate services, you will reuse it across different stages of the application.

**Reply by omar-dulaimi** [reply]
This seems promising. Anywhere I could find a working shared stages example?

**Reply by pgrzesik** [reply]
Sadly, I don't have any examples of such setup

**Reply by akshitkathuria** [reply]
@omar-dulaimi did you find any working example of it or did you implement the same. I want to implement the same but am stuck.

**Reply by mnapoli** [reply]
Just a word of caution: this is a setup that can end up being a bit more complex.

If it is acceptable for your project, it is much more recommended to go with the default approach (where all AWS resources are duplicated for every stage).

**Reply by AshKemp23** [reply]
Could anyone provide an example for deploying multiple stages in the same API gateway through serverless?
